service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/HttpLandingZoneService.java

```diff
@@ -0,0 +1,265 @@
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  @Override
+  public ApiDeleteAzureLandingZoneResult startLandingZoneDeletionJob(
+      BearerToken bearerToken, String jobId, UUID landingZoneId, String resultEndpoint)
+      throws InterruptedException {
+    var client = getLandingZonesApi(bearerToken);
+    var body = new DeleteAzureLandingZoneRequestBody().jobControl(new JobControl().id(jobId));
+    try {
+      var result = LzsRetry.retry(() -> client.deleteAzureLandingZone(body, landingZoneId));
+      return typeAdapter.toApiDeleteAzureLandingZoneResultFromApiClient(result);
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  @Override
+  public ApiDeleteAzureLandingZoneJobResult getDeleteLandingZoneResult(
+      BearerToken bearerToken, UUID landingZoneId, String jobId) throws InterruptedException {
+    var client = getLandingZonesApi(bearerToken);
+    try {
+      var result =
+          LzsRetry.retry(() -> client.getDeleteAzureLandingZoneResult(landingZoneId, jobId));
+      return typeAdapter.toApiDeleteAzureLandingZoneJobResultFromApiClient(result);
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  @Override
+  public ApiAzureLandingZoneResult getAsyncJobResult(BearerToken bearerToken, String jobId)
+      throws InterruptedException {
+    var client = getLandingZonesApi(bearerToken);
+    try {
+      var result = LzsRetry.retry(() -> client.getCreateAzureLandingZoneResult(jobId));
+      return typeAdapter.toApiAzureLandingZoneResultFromApiClient(result);
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  @Override
+  public ApiAzureLandingZone getAzureLandingZone(BearerToken bearerToken, UUID landingZoneId)
+      throws InterruptedException {
+    var client = getLandingZonesApi(bearerToken);
+    try {
+
+      var result = LzsRetry.retry(() -> client.getAzureLandingZone(landingZoneId));
+      return typeAdapter.toApiAzureLandingZone(result);
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  @Override
+  public ApiAzureLandingZoneList listLandingZonesByBillingProfile(
+      BearerToken bearerToken, UUID billingProfileId) throws InterruptedException {
+    var client = getLandingZonesApi(bearerToken);
+    ApiAzureLandingZoneList result = new ApiAzureLandingZoneList();
+
+    try {
+      var response = LzsRetry.retry(() -> client.listAzureLandingZones(billingProfileId));
+      var landingZones = response.getLandingzones();
+      landingZones.forEach(
+          landingZone ->
+              result.addLandingzonesItem(typeAdapter.toApiAzureLandingZone(landingZone)));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+
+    return result;
+  }
+
+  @Override
+  public ApiAzureLandingZoneList listLandingZones(BearerToken bearerToken) {
+    throw new RuntimeException("todo");
```

Comment:
This is problematic, it looks like Leo is using this endpoint for some reason. I'm investigating a resolution.